addons/website_sale/models/product_image.py

```diff
@@ -41,3 +41,16 @@ def _check_valid_video_url(self):
         for image in self:
             if image.video_url and not image.embed_code:
                 raise ValidationError(_("Provided video URL for '%s' is not valid. Please enter a valid video URL.") % image.name)
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        new_context = self.with_context({k: v for k, v in self.env.context.items() if k != 'default_product_tmpl_id'})
+        recs = []
+
+        for vals in vals_list:
+            if vals.get('product_variant_id') and 'default_product_tmpl_id' in self.env.context:
+                recs.append(super(ProductImage, new_context).create(vals))
```

Comment:
Note: there is actually a "complex" issue with this in that the resulting recordset will not be ordered the same way as the input values.

I still think the performance gain is worth it though, especially since the mixed use case is not a real one. There is never a case where the user would have `default_product_tmpl_id` in his context while setting `product_variant_id` on some images and not setting it on others. He's either editing a template or a variant, not both together.